examples/connext_dds/waitset_status_cond/py/waitset_status_cond_subscriber.py

```diff
@@ -0,0 +1,92 @@
+# (c) Copyright, Real-Time Innovations, 2024.  All rights reserved.
+# RTI grants Licensee a license to use, modify, compile, and create derivative
+# works of the software solely for use with RTI Connext DDS. Licensee may
+# redistribute copies of the software provided that all such copies are subject
+# to this license. The software is provided "as is", with no warranty of any
+# type, including any warranty for fitness for any purpose. RTI is under no
+# obligation to maintain or support the software. RTI shall not be liable for
+# any incidental or consequential damages arising out of the use or inability
+# to use the software.
+
+import argparse
+
+import rti.connextdds as dds
+
+from foo import Foo
+
+
+def subscriber_main(domain_id, sample_count):
+    # Create a DomainParticipant with default QoS
+    participant = dds.DomainParticipant(domain_id)
+
+    # Create a Topic and automatically register the type
+    topic = dds.Topic(participant, "Example waitset_status_cond", Foo)
+
+    # Create a DataReader with default QoS (Subscriber created in-line)
+    reader = dds.DataReader(dds.Subscriber(participant), topic)
+
+    # Create a Status Condition for the reader
+    status_condition = dds.StatusCondition(reader)
+
+    # Enable statuses configuration for the Status Condition
+    status_condition.enabled_statuses = (
+        dds.StatusMask.LIVELINESS_CHANGED | dds.StatusMask.DATA_AVAILABLE
+    )
+
+    # This counter will keep track of the samples we have processed
+    samples_read = 0
+
+    # Define a handler for the status_condition
+    def status_handler(_):
+        status_mask = reader.status_changes
+
+        # Handle liveliness status changes
+        if dds.StatusMask.LIVELINESS_CHANGED in status_mask:
+            st = reader.liveliness_changed_status
+            print(f"Liveliness changed => Active writers = {st.alive_count}")
+
+        # Handle when there's new data available
+        if dds.StatusMask.DATA_AVAILABLE in status_mask:
+            nonlocal samples_read
+            for data, info in reader.take():
```

Comment:
```suggestion
            for data in reader.take_data():
```

and remove he `if info.valid`.